synapse/tools/layer/dump.py

```diff
@@ -0,0 +1,170 @@
+            mesg = f'Layer dump tool only works on cortexes, not {celltype}'
+            raise s_exc.TypeMismatch(mesg=mesg)
+
+        celliden = info['cell']['iden']
+        cellvers = info['cell']['version']
+
+    # Find and read state file
+    state = {}
+    statefile = opts.statefile
+    if statefile is None:
+        statefile = s_common.genpath(opts.outdir, f'{celliden}.{opts.iden}.yaml')
+
+    if (data := s_common.yamlload(statefile)) is not None:
+        state = data
+
+    if (soffs := opts.offset) is None:
+        soffs = state.get('offset:next', 0)
+
+    eoffs = None
+
+    async with await s_telepath.openurl(opts.url, name=f'*/layer/{opts.iden}') as layer:
+
+        # Handle no edits to export
+        if soffs >= await layer.getEditIndx():
+            mesg = f'No edits to export starting from offset ({soffs})'
+            raise s_exc.BadArg(mesg=mesg)
+
+        finished = False
+
+        genr = layer.syncNodeEdits2(soffs, wait=False)
+
+        nodeiter = aiter(genr)
+
+        while not finished:
+
+            kwargs = {}
+
+            if s_common.majmin >= (3, 12): # pragma: no cover
+                kwargs['delete_on_close'] = False
+
+            with tempfile.NamedTemporaryFile(dir=opts.outdir, delete=False, **kwargs) as fd:
+
+                try:
+                    # Pull the first edit so we can get the starting offset
+                    first = await anext(nodeiter)
+                except StopAsyncIteration: # pragma: no cover
+                    break
+
+                soffs = first[0]
+
+                count = 1
+
+                # Write header to file
+                fd.write(s_msgpack.en((
+                    'init',
+                    {
+                        'hdrvers': 1,
+                        'celliden': celliden,
+                        'cellvers': cellvers,
+                        'layriden': opts.iden,
+                        'offset': soffs,
+                        'chunksize': opts.chunksize,
+                        'tick': s_common.now(),
+                    }
+                )))
+
+                # Now write the first edit that we already pulled
+                fd.write(s_msgpack.en(('edit', first)))
+
+                async for nodeedit in nodeiter:
+
+                    # Write individual edits to file
+                    fd.write(s_msgpack.en(('edit', nodeedit)))
+
+                    eoffs = nodeedit[0]
+
+                    count += 1
+
+                    if count % opts.chunksize == 0:
+                        break
+
+                else:
+                    finished = True
+
+                # Write footer to file
+                fd.write(s_msgpack.en(('fini', {
+                    'offset': eoffs,
+                    'tock': s_common.now(),
+                })))
+
+                tmpname = fd.name
+
+            path = os.path.join(opts.outdir, f'{celliden}.{opts.iden}.{soffs}-{eoffs}.nodeedits')
+            os.rename(tmpname, path)
```

Comment:
This assumes the dest filesystem is the same as the one used by the tempfile right?

Reply:
It 100% is because the tempfile is created in the outdir